pennylane/math/quantum.py

```diff
@@ -863,40 +730,19 @@ def sqrt_matrix(density_matrix):
         (tensor_like): Square root of the density matrix.
     """
     evs, vecs = qml.math.linalg.eigh(density_matrix)
-    evs = np.real(evs)
+    evs = qml.math.real(evs)
     evs = qml.math.where(evs > 0.0, evs, 0.0)
     if not is_abstract(evs):
         evs = qml.math.cast_like(evs, vecs)
 
     shape = qml.math.shape(density_matrix)
     if len(shape) > 2:
         # broadcasting case
-        sqrt_evs = qml.math.expand_dims(qml.math.sqrt(evs), 1) * qml.math.eye(shape[-1])
+        i = qml.math.cast_like(qml.math.convert_like(qml.math.eye(shape[-1]), evs), evs)
+        sqrt_evs = qml.math.expand_dims(qml.math.sqrt(evs), 1) * i
         return vecs @ sqrt_evs @ qml.math.conj(qml.math.transpose(vecs, (0, 2, 1)))
 
-    return vecs @ qml.math.diag(np.sqrt(evs)) @ np.conj(np.transpose(vecs))
-
-
-def _compute_fidelity(density_matrix0, density_matrix1):
-    r"""Compute the fidelity for two density matrices with the same number of wires.
-
-    .. math::
-            F( \rho , \sigma ) = -\text{Tr}( \sqrt{\sqrt{\rho} \sigma \sqrt{\rho}})^2
-    """
-    # Implementation in single dispatches (sqrt(rho))
-    sqrt_mat = qml.math.sqrt_matrix(density_matrix0)
-
-    # sqrt(rho) * sigma * sqrt(rho)
-    sqrt_mat_sqrt = sqrt_mat @ density_matrix1 @ sqrt_mat
-
-    # extract eigenvalues
-    evs = qml.math.eigvalsh(sqrt_mat_sqrt)
-    evs = np.real(evs)
-    evs = qml.math.where(evs > 0.0, evs, 0.0)
-
-    trace = (qml.math.sum(qml.math.sqrt(evs), -1)) ** 2
-
-    return trace
+    return vecs @ qml.math.diag(qml.math.sqrt(evs)) @ np.conj(np.transpose(vecs))

```

Comment:
Not sure if this is a mistake.

```suggestion
    return vecs @ qml.math.diag(qml.math.sqrt(evs)) @ qml.math.conj(qml.math.transpose(vecs))
```

Reply:
Ah good catch 😄